New package: golang-migrate-4.18.1

lpww opened the pull request:
### Description
This pr adds a new package, `golang-migrate v4.18.1`. It builds upon a [previous pull request](https://github.com/void-linux/void-packages/pull/30634) that works but was never merged. I just updated the version, checksum, and maintainer.

#### Testing the changes
- I tested the changes in this PR: **YES**

#### New package
- This new package conforms to the [package requirements](https://github.com/void-linux/void-packages/blob/master/CONTRIBUTING.md#package-requirements): **YES**

#### Local build testing
- I built this PR locally for my native architecture, **x86_64-glibc**

Bnyro commented on srcpkgs/golang-migrate/template:
```diff
@@ -0,0 +1,22 @@
+pkgname=golang-migrate
+version=4.17.1
+revision=1
+wrksrc="migrate-${version}"
```
It does build as intended without that line as well, so it should be dropped.

lpww replied:
Sorry, I'm just seeing this comment. Thanks for the feedback, I will test it and update the pr!

lpww replied:
@Bnyro I've updated the pr to remove the `wrksrc` field. Please re-review this when you get a chance

lpww commented:
@classabbyamp Please may you give this a maintainer review when you have a chance?

lpww commented:
Bumping to prevent this relevant pr from being closed

classabbyamp commented on srcpkgs/golang-migrate/template:
```diff
@@ -0,0 +1,21 @@
+checksum="59a92e27d58f22ce3270086d385b3548a2a44e09f91b0eda3e245f94bcf8f081"
+
+do_build() {
+	VERSION="$version" make build
```
don't use the makefile, emulate it with these variables: https://github.com/void-linux/void-packages/blob/master/Manual.md#go-packages

lpww replied:
Thanks for the feedback! I have refactored the template to make use of the go package options

classabbyamp commented on srcpkgs/golang-migrate/template:
```diff
@@ -0,0 +1,21 @@
+# Template file for 'golang-migrate'
+pkgname=golang-migrate
```
per [guidelines](https://github.com/void-linux/void-packages/blob/master/Manual.md#programs), use the repo name, `migrate`

lpww replied:
My bad! I will update this

classabbyamp commented on srcpkgs/golang-migrate/template:
```diff
@@ -0,0 +1,21 @@
+	VERSION="$version" make build
+}
+
+do_install() {
+	vbin migrate
```
```suggestion
post_install() {
```

lpww commented:
@classabbyamp Thanks for taking the time to review this pr. I have addressed your comments and also updated the template to use the latest version of `golang-migrate`!

classabbyamp commented:
next time, this needs to be squashed into a single commit

lpww commented:
Thanks @classabbyamp, I appreciate you taking the time to look at this! I will make sure my prs only contain a single commit in the future
